.github/workflows/yocto-build-deploy.yml

```diff
@@ -640,14 +640,29 @@ jobs:
       # hostapp Deploy
       ##############################
 
+      # Separate this evaluation into its own step + output, as we use this logic in several places and its easier to manage this way
+      - name: Evaluate whether to finalize release
+        if: steps.merge-test-result.outputs.finalize || inputs.force-finalize
+        id: should-finalize
+        run: |
+          echo "finalize=true" >> $GITHUB_OUTPUT
+
+      # Separate this evaluation into its own step + output, as we use this logic in several places and its easier to manage this way
+      # We want to push a hostapp on push events (PR merge) , or dispatch??
+      - name: Evaluate whether to deploy hostapp
+        if: github.event_name == 'push' || github.event_name == 'workflow_dispatch'
+        id: should-deploy
+        run: |
+          echo "deploy=true" >> $GITHUB_OUTPUT
+
       - name: Check Balena CLI installation
         run: |
           balena --version
 
       # TODO: replace this with balena-io/deploy-to-balena-action when it supports deploy-only
       # https://github.com/balena-io/deploy-to-balena-action/issues/286
       - name: Deploy to balena
-        if: inputs.deploy-hostapp == true
+        if: steps.should-deploy.outputs.deploy
```

Comment:
Does this actually resolve to true/false based on the value being set/unset? Very handy if it does!

Reply:
I think it does, was based on this:

"Note that in conditionals, falsy values (false, 0, -0, "", '', null) are coerced to false and truthy (true and other non-falsy values) are coerced to true."

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/expressions#literals